### PR TITLE
First draft of guides, about, faq; relocation examples

### PR DESCRIPTION
**Pull request by koenhelwegen**
A first draft of the updated documentation. I still want to proofread everything myself, but curious to hear your thoughts on the general direction.

The main thing I would really love to add is tensorboard results to illustrate various points (e.g. real-valued first/last layer, different optimizers, etcetera). I'm not sure if it is feasible to include this now though.

### CODE REVIEW

File: docs/architecture_guide.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+Here you will find a quick overview of the best practices that have evolved in the BNN community over the past few years regarding BNN architecture. After following this guide, you should be able to start designing a BNN for your application of interest that is both efficient and powerful.
+
+## Binarizing a Single Layer
+
+Any layer has two types of inputs: parameters, such as filter weights and biases, and incoming activations.
```
Comment (MariaHeuss) [inline docs/architecture_guide.md]: missing ',' after filter

File: docs/faq.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  url          = {https://larq.dev},
+  year         = {2019}
+}
+```
+
+If your paper is publicly available, feel free to also add it to the list of [Papers using Larq](/papers).
+
+## Can I add my own algorithm or model to Larq?
+
+Absolutely! If you have developed a new BNN or training method that you would like to share with the community, create a PR or get in touch with us. Make sure you check out the contribution guide. For entire models with pretrained weights [`larq-zoo`](https://github.com/plumerai/larq-zoo) is the correct place, everything else can be added to [`larq`](https://github.com/plumerai/larq) directly.
```
Comment (MariaHeuss) [inline docs/faq.md]: Here I would also mention that larq is not only meant for BNNs but also for other quantized networks and that code also for those networks as well as models are welcome as well.

File: docs/guide.md [hunk 3]
```diff
@@ -1,8 +1,22 @@
...
+If you're new to Larq and/or BNNs, this is the right place to start. Below we summarize the key concepts you need to understand to work with BNNs.
 
-Each quantized layer requires an `input_quantizer` and a `kernel_quantizer` that describe the way of quantizing the incoming activations and weights of the layer respectively. If both `input_quantizer` and `kernel_quantizer` are `None` the layer is equivalent to a full precision layer.
+## Quantizer
+
+The core idea of BNNs is to use binary values instead of 32-bit floating point. A [quantizer](https://larq.dev/api/quantizers/) defines the way of transforming a full precision input to a quantized output and the pseudo-gradient method used for the backwards pass.
+
+Note that most layers in a neural network evaluate sums and thus output integers even if all inputs are binary. Therefore you will usually want to apply quantizers for your activations throughout the network even during inference.
```
[the remark this answers is not in the document]
Comment (MariaHeuss) [reply docs/guide.md]: Agreed

File: docs/guide.md [hunk 4]
```diff
@@ -1,8 +1,22 @@
 # User Guide
 
-To build a Quantized Neural Network (QNN), Larq introduces the concept of [quantized layers](https://larq.dev/api/layers/) and [quantizers](https://larq.dev/api/quantizers/). A quantizer defines the way of transforming a full precision input to a quantized output and the pseudo-gradient method used for the backwards pass.
+If you are new to Larq and/or Binarized Neural Networks (BNNs), this is the right place to start. Below we summarize the key concepts you need to understand to work with BNNs.
 
-Each quantized layer requires an `input_quantizer` and a `kernel_quantizer` that describe the way of quantizing the incoming activations and weights of the layer respectively. If both `input_quantizer` and `kernel_quantizer` are `None` the layer is equivalent to a full precision layer.
+## Quantizer
+
+The core idea of BNNs is to use binary weights and activations instead of 32-bit floating point values. A [quantizer](https://larq.dev/api/quantizers/) defines the way of transforming a full precision input to a quantized output and the pseudo-gradient method used for the backwards pass.
```
Comment (MariaHeuss) [inline docs/guide.md]: I would add "during the forward pass" to avoid confusion about latent weights.

File: docs/guide.md [hunk 5]
```diff
@@ -1,8 +1,22 @@
...
+
+Note that most layers in a neural network evaluate sums and thus output integers even if all inputs are binary. Therefore you will usually want to apply quantizers for your activations throughout the network even during inference.
+
+It is also common to apply quantizers to the weights during training. This is necessary when relying on real-valued latent-weights to accumulate non-binary update steps, a common optimization strategy for BNNs. After training is finished, the real-valued weights and associated quantization operations can be discarded.
```
Comment (MariaHeuss) [inline docs/guide.md]: I think this whole paragraph is a bit confusing. The idea with the latent weights in not super intuitive therefore I would stress this a bit more and explicitly say that you binarize the weights during each forward pass.

File: docs/optimization_guide.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
+Once you have defined a good architecture for your BNN, you want to train it. Here we give an introduction to common training strategies and tricks that are popular in the field. After reading this guide you should have a good idea of how you can train your BNN.
+
+Below we first discuss the fundamental challenges of BNN optimization. We then explain the most common used training strategy, of using latent weights, and cover the questions, tips & tricks that tend to come up when training BNNs.
```
Comment (MariaHeuss) [inline docs/optimization_guide.md]: commonly

File: docs/architecture_guide.md [hunk 7]
```diff
@@ -12,6 +12,17 @@ Note that the output of a binarized layer is _not_ binary. Instead the output is
 
 When viewing binarization as an activation function just like ReLU, one may be inclined to binarize the outgoing activations rather than the incoming activations. However, if the network contains batch normalization layers or residual connections, this may result in unintentional non-binary operations. Therefore we have opted for an `input_quantizer` rather than an `activation_quantizer`.
 
+A typical binarized layer looks something like:
+
+```python
+x_out = lq.layers.QuantDense(
```
Comment (lgeiger) [inline docs/architecture_guide.md]: I think we should either `import  larq as lq` somewhere or consistently use `larq` instead of `lq`.

File: docs/architecture_guide.md [hunk 8]
```diff
@@ -34,10 +45,86 @@ An issue with ResNet-style shortcuts comes up when there is a dimensionality cha
...
+            padding="same",
+            )(shortcut)
+    if in_filters != out_filters:
+        shortcut = tf.keras.layers.Conv2D(
+            out_filters,
+            1,
+            kernel_initializer="glorot_normal",
+            use_bias=False,
+        )(shortcut)
+        shortcut = tf.keras.layers.BatchNormalization(momentum=0.9)(shortcut)
```
Comment (lgeiger) [inline docs/architecture_guide.md]: I personally would remove the handling for changing the dimension in the, to make this example easier to understand. Since there are many ways how you could handle dimension changes I don't think we should add this here as an example.

File: docs/optimization_guide.md [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+Instead of using latent weights, one can opt for a custom BNN optimizer that inherently generates binary weights. An example of such an optimizer is [Bop](api/optimizers/#bop).
+
+## Choice of Pseudo-Gradient
+
+In [`larq.quantizers`](/api/quantizers) you will find a variety of quantizers that have been introduced in different papers. Many of these quantizers behave identically during the forward pass but implement different pseudo-gradients. Studies comparing different pseudo-gradients report little difference between them. Therefore, we recommend using the classical [`ste_sign()`](/api/quantizers/#ste_sign) as default.
+
+## Choice of Optimizer
+
+When using a latent weight strategy, you can apply any optimizer you are familiar with from real-valued deep learning. However, due the different nature of BNNs your intuitions may be off. We recommend using Adam: although other optimizers can achieve similar accuracies with a lot of finetuning, we and others have found that Adam is easiest to use for most types of networks.
+
```
Comment (jamescook106) [inline docs/optimization_guide.md]: I would say that Adam is quickest to converge and least sensitive to the choice of hyperparamters

File: docs/optimization_guide.md [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+## Tips & Tricks
+
+Here are some general tips and tricks that you may want to keep in mind:
+
+- BNN training is more noisy due to non-continuous nature of flipping weights; therefore, we recommend setting your batch norm momentum to 0.9.
+- Similarly, we recommend using a high batch size (e.g. 512, 1024 or even higher).
```
Comment (jamescook106) [inline docs/optimization_guide.md]: Not sure this is true, binarynet on CIFAR10 uses batch size 50
